### PR TITLE
fix upstream helm support

### PR DESCRIPTION
**Pull request by jrryjcksn**
**What changed?**
We now generate the correct source type for upstream helm and we disallow using it in the user repo mode.

**Why?**
Upstream helm did not work due to an oversight when merging in change.

**How did you test it?**
Tested locally

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: pkg/cmdimpl/add.go [hunk 1]
```diff
@@ -88,6 +88,11 @@ func updateParametersIfNecessary(gitClient git.Git) error {
 		params.DeploymentType = string(DeployTypeHelm)
 	}
 
+	if params.AppConfigUrl == "" && params.SourceType != string(SourceTypeGit) {
```
Comment (palemtnrider) [inline pkg/cmdimpl/add.go:91]: We should use `string(ConfigTypeUserRepo)` instead of the empty string.

File: pkg/cmdimpl/add.go [hunk 2]
```diff
@@ -88,6 +88,11 @@ func updateParametersIfNecessary(gitClient git.Git) error {
...
+		return fmt.Errorf("Cannot create .wego directory in helm repository:\n" +
```
Comment (palemtnrider) [inline pkg/cmdimpl/add.go:92]: Nit: error message should start with a lower case letter (go style guide)

**Review by palemtnrider — APPROVED** [review]
Other than a minor change and a nit, LGTM

File: pkg/cmdimpl/add.go (hunk 2 above)
Comment (jrryjcksn) [reply pkg/cmdimpl/add.go:92]: Good catch

**Pull request merged**